src/unstract/sdk/tool/validator.py

```diff
@@ -211,8 +208,12 @@ def _validate_file_type(self, input_file: Path) -> None:
                     f"{ext} mentioned in tool PROPERTIES is not supported"
                 )
             allowed_mimes.append(EXT_MIME_MAP[ext])
-
-        input_file_mime = ToolUtils.get_file_mime_type(input_file=input_file)
+        if self.tool.workflow_filestorage:
+            input_file_mime = ToolUtils.get_file_mime_type(
```

Comment:
@muhammad-ali-e We can use [this](https://github.com/Zipstack/unstract-sdk/blob/ef00f8149f19c586bd967b8cb0dae6965dc9e1e8/src/unstract/sdk/file_storage/fs_impl.py#L219) method to fetch `mime_type`
We can completely remove `get_file_mime_type`, and replace with the new implementation, leveraging the new methods. 
CC : @gaya3-zipstack

Reply:
Pls take care of this in the next PR @muhammad-ali-e